pennylane/clusters.py

```diff
@@ -0,0 +1,476 @@
+            final step of each circuit
+        device (Device, List[Device]): Corresponding device(s) where the resulting
+            QNodeCluster should be executed. This can either be a single device, or a list
+            of devices of length ``len(observables)``.
+        measure (str, List[str]): Measurement(s) to perform. Options include
+            :func:`'expval' <pennylane.expval>`, :func:`'var' <pennylane.var>`,
+            and :func:`'sample' <pennylane.sample>`.
+            This can either be a single measurement type, in which case it is applied
+            to all observables, or a list of measurements of length ``len(observables)``.
+        interface (str, None): which interface to use for the returned QNode cluster.
+            This affects the types of objects that can be passed to/returned from the QNode:
+
+            * ``interface='autograd'``: Allows autograd to backpropogate
+              through the QNodeCluster. The QNodeCluster accepts default Python types
+              (floats, ints, lists) as well as NumPy array arguments,
+              and returns NumPy arrays.
+
+            * ``interface='torch'``: Allows PyTorch to backpropogate
+              through the QNodeCluster.The QNodeCluster accepts and returns Torch tensors.
+
+            * ``interface='tf'``: Allows TensorFlow in eager mode to backpropogate
+              through the QNodeCluster. The QNodeCluster accepts and returns
+              TensorFlow ``tf.Variable`` and ``tf.tensor`` objects.
+
+            * ``None``: The QNodeCluster accepts default Python types
+              (floats, ints, lists) as well as NumPy array arguments,
+              and returns NumPy arrays. It does not connect to any
+              machine learning library automatically for backpropagation.
+
+        diff_method (str, None): the method of differentiation to use in the created QNodeCluster.
+
+            * ``"best"``: Best available method. Uses the device directly to compute
+              the gradient if supported, otherwise will use the analytic parameter-shift
+              rule where possible with finite-difference as a fallback.
+
+            * ``"parameter-shift"``: Use the analytic parameter-shift
+              rule where possible with finite-difference as a fallback.
+
+            * ``"finite-diff"``: Uses numerical finite-differences.
+
+            * ``None``: a non-differentiable QNodeCluster is returned.
+
+    Returns:
+        QNodeCluster: a cluster of QNodes executing the circuit template with
+        the specified measurements
+
+    **Example:**
+
+    Let's define a custom two-wire template:
+
+    .. code-block:: python
+
+        def my_template(params, wires, **kwargs):
+            for i in range(2):
+                qml.RX(params[i], wires=wires[i])
+
+            qml.CNOT(wires=wires)
+
+    We want to compute the expectation values over the following list
```

Comment:
Would adding an example for returning the ``probabilities`` make this section too big (since it's already quite detailed)? It might just be not so clear that once has to pass an empyt list for the ``observables``

Reply:
In the first commit I made, I had an `if` statement to make this work for probabilities. But I removed it, for two reasons:

1. As per the ADR, the first version of this functionality should be as bare bones as possible, and

2. The use-case for returning `probs` is not as clear, since every QNode will then have the same ansatz _and_ the same return statement. This isn't really a mapping, so might as well just instantiate the QNode cluster manually.

Reply:
Gotcha! I had the impression that the ``probs`` still stayed